src/io/platform/f4xx/GPIOf4xx.cpp

```diff
@@ -0,0 +1,213 @@
+#include <EVT/io/platform/f4xx/GPIOf4xx.hpp>
+
+#include <stdint.h>
+
+#include <EVT/io/GPIO.hpp>
+#include <EVT/io/pin.hpp>
+
+#include "EVT/io/platform/f4xx/GPIOf4xx.hpp"
+#include <EVT/platform/f4xx/stm32f4xx.hpp>
+#include <HALf4/stm32f4xx_hal_gpio.h>
+#include <HALf4/stm32f4xx_hal_rcc.h>
+
+void (*INTERRUPT_HANDLERS[16])(EVT::core::IO::GPIO* pin) = {nullptr};
+EVT::core::IO::GPIO* INTERRUPT_GPIOS[16] = {nullptr};
+
+extern "C" void EXTI0_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_0);
+}
+extern "C" void EXTI1_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_1);
+}
+extern "C" void EXTI2_TSC_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_2);
+}
+extern "C" void EXTI3_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_3);
+}
+extern "C" void EXTI4_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_4);
+}
+extern "C" void EXTI9_5_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_5);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_6);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_7);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_8);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_9);
+}
+
+extern "C" void EXTI15_10_IRQHandler(void) {
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_10);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_11);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_12);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_13);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_14);
+    HAL_GPIO_EXTI_IRQHandler(GPIO_PIN_15);
+}
+
+extern "C" void HAL_GPIO_EXTI_Callback(uint16_t GPIO_Pin) {
+    uint16_t tmpPin = GPIO_Pin;
+    uint8_t count = 0;
+
+    while ((tmpPin & 0x0001) == 0) {
+        tmpPin = tmpPin >> 1;
+        count++;
+    }
+    if (nullptr != INTERRUPT_HANDLERS[count]) {
+        INTERRUPT_HANDLERS[count](INTERRUPT_GPIOS[count]);
+    }
+}
+
+namespace EVT::core::IO {
+
+GPIOf4xx::GPIOf4xx(Pin pin, GPIO::Direction direction)
+    : GPIO(pin, direction) {
+
+    GPIO_InitTypeDef gpioInit;
+    Pin myPins[] = {pin};
+    uint8_t numOfPins = 1;
+
+    gpioStateInit(&gpioInit, myPins, numOfPins,
+                  static_cast<uint32_t>(direction), GPIO_PULLDOWN, GPIO_SPEED_FREQ_HIGH);
+
+    switch ((static_cast<uint8_t>(pin) & 0xF0) >> 4) {
+    case 0x0:
```

Comment:
Add a TODO to double check that this actually applies to F4xx the same way it does to F3xx